--- hppc-core/pom.xml	2013-03-01 12:06:26.000000000 +0100
+++ hppc-core/pom.xml-gil	2013-08-25 14:48:26.793151962 +0200
@@ -101,41 +101,6 @@
             </goals>
           </execution>
           
-          <execution>
-            <id>antrun-retrotranslate-jdk15</id>
-            <phase>package</phase>
-            <configuration>
-              <target>
-                <taskdef name="retrotranslator"
-                  classname="net.sf.retrotranslator.transformer.RetrotranslatorTask"
-                  classpathref="maven.plugin.classpath" />
-
-                <retrotranslator verbose="true" verify="false"
-                  failonwarning="true" target="1.5" embed="com.carrotsearch.hppc.backport"
-                  destjar="${project.build.directory}/${project.build.finalName}-jdk15.jar">
-
-                  <jarfileset dir="${project.build.directory}"
-                    includes="${project.build.finalName}.jar" />
-
-                  <classpath>
-                    <pathelement location="${basedir}/src/main/retrotranslator" />
-                  </classpath>
-                  <classpath refid="maven.plugin.classpath" />
-                  <classpath refid="maven.compile.classpath" />
-                  <classpath>
-                    <fileset dir="${java.home}/lib" includes="rt.jar" />
-                  </classpath>
-                </retrotranslator>
-
-                <attachartifact
-                  file="${project.build.directory}/${project.build.finalName}-jdk15.jar"
-                  classifier="jdk15" type="jar" />
-              </target>
-            </configuration>
-            <goals>
-              <goal>run</goal>
-            </goals>
-          </execution>
         </executions>
 
         <dependencies>
@@ -144,24 +109,7 @@
             <artifactId>hppc-templateprocessor</artifactId>
             <version>${project.version}</version>
           </dependency>
-          
-          <dependency>
-            <groupId>net.sf.retrotranslator</groupId>
-            <artifactId>retrotranslator-transformer</artifactId>
-            <version>${retrotranslator.version}</version>
-          </dependency>
 
-          <dependency>
-            <groupId>net.sf.retrotranslator</groupId>
-            <artifactId>retrotranslator-runtime</artifactId>
-            <version>${retrotranslator.version}</version>
-          </dependency>
-
-          <dependency>
-            <groupId>backport-util-concurrent</groupId>
-            <artifactId>backport-util-concurrent</artifactId>
-            <version>${backport-util-concurrent.version}</version>
-          </dependency>
         </dependencies>
       </plugin>
 
