diff -ur logback-0.9.18.orig/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java logback-0.9.18/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java --- logback-0.9.18.orig/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java	2010-01-11 13:37:31.360869985 +0000
+++ logback-0.9.18/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java	2010-01-11 14:04:54.953081269 +0000
@@ -251,4 +251,10 @@
   public Filter getFirstFilter() {
     return fai.getFirstFilter();
   }
+
+  public void addLifeCycleListener(org.mortbay.component.LifeCycle.Listener listener) {
+  }
+
+  public void removeLifeCycleListener(org.mortbay.component.LifeCycle.Listener listener) {
+  }
 }
