diff -Nru logback-1.1.2/logback-access/pom.xml logback-1.1.2.jetty/logback-access/pom.xml
--- logback-1.1.2/logback-access/pom.xml	2014-04-02 14:10:03.000000000 +0200
+++ logback-1.1.2.jetty/logback-access/pom.xml	2015-03-27 12:42:41.069669011 +0100
@@ -59,6 +59,13 @@
 
     <dependency>
       <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-util</artifactId>
+      <scope>compile</scope>
+      <optional>true</optional>
+    </dependency>
+
+    <dependency>
+      <groupId>org.eclipse.jetty</groupId>
       <artifactId>jetty-server</artifactId>
       <scope>compile</scope>
       <optional>true</optional>
diff -Nru logback-1.1.2/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java logback-1.1.2.jetty/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java
--- logback-1.1.2/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java	2013-09-07 12:44:46.000000000 +0200
+++ logback-1.1.2.jetty/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java	2015-03-27 12:53:44.091008201 +0100
@@ -27,6 +27,7 @@
 import org.eclipse.jetty.server.Request;
 import org.eclipse.jetty.server.RequestLog;
 import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.component.LifeCycle;
 
 
 import ch.qos.logback.access.joran.JoranConfigurator;
@@ -293,11 +294,15 @@
     return fai.getFilterChainDecision(event);
   }
 
-  public void addLifeCycleListener(Listener listener) {
+  public void addLifeCycleListener(LifeCycle.Listener  listener) {
     // we'll implement this when asked
   }
 
-  public void removeLifeCycleListener(Listener listener) {
+  public void removeLifeCycleListener(LifeCycle.Listener  listener) {
+    // we'll implement this when asked
+  }
+  
+  public void log(Request request, int status, long written) {
     // we'll implement this when asked
   }
 
