--- logback-access/pom.xml	2012-07-21 12:57:52.649083960 +0200
+++ logback-access/pom.xml-gil	2012-07-21 12:58:06.398383589 +0200
@@ -48,6 +48,16 @@
       <artifactId>tomcat-catalina</artifactId>
       <scope>compile</scope>
       <optional>true</optional>
+      <exclusions>
+        <exclusion>
+          <groupId>org.apache.tomcat</groupId>
+          <artifactId>tomcat-api</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.apache.tomcat</groupId>
+          <artifactId>tomcat-util</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
 
     <!--<dependency>-->
