--- logback-core/src/main/java/ch/qos/logback/core/joran/conditional/PropertyEvalScriptBuilder.java	2011-10-28 15:12:33.000000000 +0200
+++ logback-core/src/main/java/ch/qos/logback/core/joran/conditional/PropertyEvalScriptBuilder.java-gil	2012-07-11 13:21:25.250779454 +0200
@@ -20,9 +20,7 @@
 
 import ch.qos.logback.core.spi.PropertyContainer;
 import org.codehaus.janino.ClassBodyEvaluator;
-import org.codehaus.janino.CompileException;
-import org.codehaus.janino.Parser.ParseException;
-import org.codehaus.janino.Scanner.ScanException;
+import org.codehaus.commons.compiler.CompileException;
 
 import ch.qos.logback.core.spi.ContextAwareBase;
 
@@ -41,7 +39,7 @@
   Map<String, String> map = new HashMap<String, String>();
 
   public Condition build(String script) throws IllegalAccessException,
-          CompileException, ParseException, ScanException, InstantiationException,
+          CompileException, InstantiationException,
           SecurityException, NoSuchMethodException, IllegalArgumentException,
           InvocationTargetException {
 
