diff -Nru logback-1.1.3/logback-access/pom.xml logback-1.1.3.jetty/logback-access/pom.xml
--- logback-1.1.3/logback-access/pom.xml	2015-03-24 10:38:37.000000000 +0100
+++ logback-1.1.3.jetty/logback-access/pom.xml	2015-08-07 14:35:51.958418270 +0200
@@ -46,6 +46,12 @@
       <optional>true</optional>
     </dependency>
     <dependency>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-util</artifactId>
+      <scope>compile</scope>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
       <groupId>org.codehaus.janino</groupId>
       <artifactId>janino</artifactId>
       <scope>compile</scope>
diff -Nru logback-1.1.3/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java logback-1.1.3.jetty/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java
--- logback-1.1.3/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java	2015-03-05 11:11:31.000000000 +0100
+++ logback-1.1.3.jetty/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java	2015-08-07 15:13:43.317872496 +0200
@@ -212,11 +212,6 @@
     started = false;
   }
 
-  @Override
-  public boolean isRunning() {
-    return started;
-  }
-
   public void setFileName(String fileName) {
     this.fileName = fileName;
   }
@@ -230,26 +225,6 @@
     return started;
   }
 
-  @Override
-  public boolean isStarting() {
-    return false;
-  }
-
-  @Override
-  public boolean isStopping() {
-    return false;
-  }
-
-  @Override
-  public boolean isStopped() {
-    return !started;
-  }
-
-  @Override
-  public boolean isFailed() {
-    return false;
-  }
-
   public boolean isQuiet() {
     return quiet;
   }
@@ -313,14 +288,7 @@
     return fai.getFilterChainDecision(event);
   }
 
-  @Override
-  public void addLifeCycleListener(Listener listener) {
+  public void log(Request request, int status, long written) {
     // we'll implement this when asked
   }
-
-  @Override
-  public void removeLifeCycleListener(Listener listener) {
-    // we'll implement this when asked
-  }
-
 }
