--- ./wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/knownhost/SingleKnownHostProvider.java.sav	2005-10-12 10:16:36.000000000 -0400
+++ ./wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/knownhost/SingleKnownHostProvider.java	2008-10-03 17:10:56.000000000 -0400
@@ -17,6 +17,7 @@
  */
 
 import com.jcraft.jsch.HostKeyRepository;
+import com.jcraft.jsch.HostKey;
 import com.jcraft.jsch.JSch;
 import com.jcraft.jsch.UserInfo;
 import org.codehaus.plexus.util.Base64;
@@ -59,6 +60,15 @@
     public void addKnownHosts( JSch sch, UserInfo userInfo )
     {
         HostKeyRepository hkr = sch.getHostKeyRepository();
-        hkr.add( host, Base64.decodeBase64( key.getBytes() ), userInfo );
+        //hkr.add( host, Base64.decodeBase64( key.getBytes() ), userInfo );
+        try
+        {
+         HostKey hostkey = new HostKey(host, Base64.decodeBase64(key.getBytes()));
+         hkr.add(hostkey, userInfo);
+        }
+        catch (Exception e)
+        {
+        e.printStackTrace();
+        }
     }
 }
