--- simplyhtml-0_16_05/src/build.xml	2012-08-22 19:18:23.000000000 +0200
+++ simplyhtml-0_16_05/src/copy of build.xml	2013-01-19 08:10:05.233284955 +0100
@@ -17,9 +17,10 @@
 	<property name="java_source_version" value="1.5" />
 	<property name="java_target_version" value="1.5" />
 	<property name="build.compiler" value="modern" />
-	<property name="jhall.jar" value="..\lib\jhall.jar" />
-	<property name="gnu-regexp.jar" value="..\lib\gnu-regexp-1.1.4.jar" />
-	<property name="classpath" value="${jhall.jar}:${gnu-regexp.jar}"/>
+	<property name="jhall.jar" value="/usr/share/java/javahelp2.jar" />
+	<property name="gnu-regexp.jar" value="/usr/share/java/gnu-regexp.jar" />
+	
+<property name="classpath" value="${jhall.jar}:${gnu-regexp.jar}"/>
 
 	<target name="build" depends="clean">
 		<mkdir dir="${build}"/>
