--- sisu-inject/org.eclipse.sisu.inject.tests/src/org/eclipse/sisu/inject/TypeArgumentsTest.java.~	2014-04-21 16:32:25.000000000 +0200
+++ sisu-inject/org.eclipse.sisu.inject.tests/src/org/eclipse/sisu/inject/TypeArgumentsTest.java	2014-09-12 08:14:58.641201053 +0200
@@ -10,6 +10,8 @@
  *******************************************************************************/
 package org.eclipse.sisu.inject;
 
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedType;
 import java.lang.reflect.GenericDeclaration;
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
@@ -362,6 +364,26 @@
                 return new Type[] { String.class };
             }
 
+            public AnnotatedType[] getAnnotatedBounds()
+            {
+                return new AnnotatedType[0];
+            }
+
+            public Annotation[] getAnnotations()
+            {
+                return new Annotation[0];
+            }
+
+            public Annotation[] getDeclaredAnnotations()
+            {
+                return new Annotation[0];
+            }
+
+            public <T extends Annotation> T getAnnotation(Class<T> annotationClass)
+            {
+                return null;
+            }
+
             public GenericDeclaration getGenericDeclaration()
             {
                 return null;
