--- sisu-plexus/org.eclipse.sisu.plexus/src/org/eclipse/sisu/plexus/PlexusAnnotatedMetadata.java
+++ sisu-plexus/org.eclipse.sisu.plexus/src/org/eclipse/sisu/plexus/PlexusAnnotatedMetadata.java
@@ -88,7 +88,7 @@ public final class PlexusAnnotatedMetadata
             return text;
         }
         // use same interpolation method as XML for sake of consistency
-        final Reader r = new InterpolationFilterReader( new StringReader( text ), variables );
+        final Reader r = new InterpolationFilterReader( new StringReader( text ), (Map) variables );
         try
         {
             return IOUtil.toString( r );
diff --git a/org.eclipse.sisu.plexus/src/org/eclipse/sisu/plexus/PlexusXmlScanner.java b/org.eclipse.sisu.plexus/src/org/eclipse/sisu/plexus/PlexusXmlScanner.java
index e04fb07..d715460 100644
--- sisu-plexus/org.eclipse.sisu.plexus/src/org/eclipse/sisu/plexus/PlexusXmlScanner.java
+++ sisu-plexus/org.eclipse.sisu.plexus/src/org/eclipse/sisu/plexus/PlexusXmlScanner.java
@@ -113,7 +113,7 @@ final class PlexusXmlScanner
         final Reader reader = ReaderFactory.newXmlReader( in );
         if ( null != variables )
         {
-            return new InterpolationFilterReader( reader, variables );
+            return new InterpolationFilterReader( reader, (Map) variables );
         }
         return reader;
     }
-- 
1.9.3

