--- wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/interactive/TraditionalUIKeyboardInteractive.java.sav	2005-10-10 18:37:16.000000000 +0200
+++ wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/interactive/TraditionalUIKeyboardInteractive.java	2005-11-04 12:41:30.000000000 +0100
@@ -16,6 +16,7 @@
  * limitations under the License.
  */
 
+import java.util.regex.Pattern;
 import com.jcraft.jsch.UIKeyboardInteractive;
 import org.apache.maven.wagon.authentication.AuthenticationInfo;
 
@@ -52,7 +53,7 @@
         String[] ret;
 
         if ( prompt.length == echo.length && prompt.length == 1 && !echo[0] &&
-            prompt[0].toLowerCase().contains( "password" ) )
+            Pattern.matches(".*password.*", prompt[0].toLowerCase() ) )
         {
 
             ret = new String[1];
