--- wagon-providers/wagon-scm/src/main/java/org/apache/maven/wagon/providers/scm/ScmWagon.java.sav	2005-11-04 09:57:43.000000000 +0100
+++ wagon-providers/wagon-scm/src/main/java/org/apache/maven/wagon/providers/scm/ScmWagon.java	2005-11-04 10:27:07.000000000 +0100
@@ -153,16 +153,16 @@
             // TODO: ewwww
             if ( new File( basedir, ".svn" ).exists() || new File( basedir, "CVS" ).exists() )
             {
-                scmManager.update( scmRepository, new ScmFileSet( basedir ), null );
+                scmManager.getProviderByRepository( scmRepository ).update( scmRepository, new ScmFileSet( basedir ), null );
             }
             else
             {
                 File lastOne = mkdirs( basedir, scmRepository );
-                scmManager.checkOut( scmRepository, new ScmFileSet( basedir ), null );
+                scmManager.getProviderByRepository( scmRepository ).checkOut( scmRepository, new ScmFileSet( basedir ), null );
 
                 if ( lastOne != null )
                 {
-                    ScmResult result = scmManager.checkIn( scmRepository,
+                    ScmResult result = scmManager.getProviderByRepository( scmRepository ).checkIn( scmRepository,
                                                            new ScmFileSet( lastOne.getParentFile(), lastOne ), null,
                                                            msg );
                     if ( !result.isSuccess() )
@@ -173,8 +173,8 @@
             }
 
             // TODO: detect if it is necessary to add first!
-            scmManager.add( scmRepository, new ScmFileSet( basedir, scmFile ) );
-            ScmResult result = scmManager.checkIn( scmRepository, new ScmFileSet( basedir, scmFile ), null, msg );
+            scmManager.getProviderByRepository( scmRepository ).add( scmRepository, new ScmFileSet( basedir, scmFile ) );
+            ScmResult result = scmManager.getProviderByRepository( scmRepository ).checkIn( scmRepository, new ScmFileSet( basedir, scmFile ), null, msg );
             if ( !result.isSuccess() )
             {
                 throw new TransferFailedException( "Unable to commit file " + result.getCommandOutput() );
@@ -214,7 +214,7 @@
             // TODO: ewwww
             if ( !new File( basedir, ".svn" ).exists() && !new File( basedir, "CVS" ).exists() )
             {
-                scmManager.add( scmRepository, new ScmFileSet( parent, basedir ) );
+                scmManager.getProviderByRepository( scmRepository ).add( scmRepository, new ScmFileSet( parent, basedir ) );
                 lastOne = basedir;
             }
         }
@@ -261,13 +261,13 @@
             // TODO: ewwww
             if ( new File( basedir, ".svn" ).exists() || new File( basedir, "CVS" ).exists() )
             {
-                scmManager.update( scmRepository, new ScmFileSet( basedir ), null );
+                scmManager.getProviderByRepository( scmRepository ).update( scmRepository, new ScmFileSet( basedir ), null );
             }
             else
             {
                 // TODO: this should be checking out a full hierachy (requires the -d equiv)
                 basedir.mkdirs();
-                scmManager.checkOut( scmRepository, new ScmFileSet( basedir ), null );
+                scmManager.getProviderByRepository( scmRepository ).checkOut( scmRepository, new ScmFileSet( basedir ), null );
             }
 
             if ( !scmFile.exists() )
